# Comparing `tmp/hardyweinbergcalculator-0.2.2.post20230611.zip` & `tmp/hardyweinbergcalculator-0.2.3.post20230613.zip`

## zipinfo {}

```diff
@@ -1,38 +1,38 @@
-Zip file size: 21625 bytes, number of entries: 36
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/
--rw-r--r--  2.0 unx     3472 b- defN 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/PKG-INFO
--rw-r--r--  2.0 unx     1074 b- defN 23-Jun-11 17:35 hardyweinbergcalculator-0.2.2.post20230611/LICENSE
--rw-r--r--  2.0 unx     2118 b- defN 23-Jun-11 17:40 hardyweinbergcalculator-0.2.2.post20230611/pyproject.toml
--rw-r--r--  2.0 unx     2135 b- defN 23-Jun-11 17:02 hardyweinbergcalculator-0.2.2.post20230611/README.md
--rw-r--r--  2.0 unx     2042 b- defN 23-Jun-11 17:37 hardyweinbergcalculator-0.2.2.post20230611/setup.py
--rw-r--r--  2.0 unx      701 b- defN 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/setup.cfg
--rw-r--r--  2.0 unx     3472 b- defN 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/PKG-INFO
--rw-r--r--  2.0 unx        1 b- defN 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/not-zip-safe
--rw-r--r--  2.0 unx     1109 b- defN 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/SOURCES.txt
--rw-r--r--  2.0 unx       62 b- defN 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/entry_points.txt
--rw-r--r--  2.0 unx       30 b- defN 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/requires.txt
--rw-r--r--  2.0 unx       24 b- defN 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/top_level.txt
--rw-r--r--  2.0 unx        1 b- defN 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/dependency_links.txt
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/genetics/
--rw-r--r--  2.0 unx      525 b- defN 23-Jun-11 17:30 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/config.py
--rw-r--r--  2.0 unx      474 b- defN 23-Jun-11 17:40 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/__init__.py
--rw-r--r--  2.0 unx     2936 b- defN 23-Jun-11 17:30 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/__main__.py
--rw-r--r--  2.0 unx     8889 b- defN 23-Jun-11 17:30 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py
--rw-r--r--  2.0 unx        0 b- defN 23-Jun-10 13:11 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/__init__.py
--rw-r--r--  2.0 unx     1750 b- defN 23-Jun-11 16:40 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/parsers/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-11 17:41 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/generators/
--rw-r--r--  2.0 unx        0 b- defN 23-Jun-11 16:40 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/__init__.py
--rw-r--r--  2.0 unx        0 b- defN 23-Jun-09 17:39 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/parsers/__init__.py
--rw-r--r--  2.0 unx     3784 b- defN 23-Jun-11 17:30 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/parsers/args_parser.py
--rw-r--r--  2.0 unx     3040 b- defN 23-Jun-11 17:30 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/generators/population_generator.py
--rw-r--r--  2.0 unx        0 b- defN 23-Jun-09 17:35 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/generators/__init__.py
--rw-r--r--  2.0 unx      372 b- defN 23-Jun-10 15:56 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/generators/char_generator.py
--rw-r--r--  2.0 unx        0 b- defN 23-Jun-10 13:11 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/genetics/__init__.py
--rw-r--r--  2.0 unx     2746 b- defN 23-Jun-11 16:40 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/genetics/gene.py
--rw-r--r--  2.0 unx     1221 b- defN 23-Jun-11 16:40 hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/genetics/allele.py
-36 files, 41978 bytes uncompressed, 13113 bytes compressed:  68.8%
+Zip file size: 21690 bytes, number of entries: 36
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/
+-rw-r--r--  2.0 unx     2904 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/PKG-INFO
+-rw-r--r--  2.0 unx     1074 b- defN 23-Jun-11 17:35 hardyweinbergcalculator-0.2.3.post20230613/LICENSE
+-rw-r--r--  2.0 unx     2114 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/pyproject.toml
+-rw-r--r--  2.0 unx     1567 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/README.md
+-rw-r--r--  2.0 unx     2038 b- defN 23-Jun-13 16:38 hardyweinbergcalculator-0.2.3.post20230613/setup.py
+-rw-r--r--  2.0 unx      697 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/setup.cfg
+-rw-r--r--  2.0 unx     2904 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/PKG-INFO
+-rw-r--r--  2.0 unx        1 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/not-zip-safe
+-rw-r--r--  2.0 unx     1109 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/SOURCES.txt
+-rw-r--r--  2.0 unx       58 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/entry_points.txt
+-rw-r--r--  2.0 unx       30 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/requires.txt
+-rw-r--r--  2.0 unx       24 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/top_level.txt
+-rw-r--r--  2.0 unx        1 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/dependency_links.txt
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/genetics/
+-rw-r--r--  2.0 unx      525 b- defN 23-Jun-11 17:30 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/config.py
+-rw-r--r--  2.0 unx      239 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/__init__.py
+-rw-r--r--  2.0 unx     2587 b- defN 23-Jun-13 21:34 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/__main__.py
+-rw-r--r--  2.0 unx     9005 b- defN 23-Jun-13 21:12 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py
+-rw-r--r--  2.0 unx       95 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/__init__.py
+-rw-r--r--  2.0 unx     1763 b- defN 23-Jun-13 20:07 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/parsers/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 21:40 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/generators/
+-rw-r--r--  2.0 unx      183 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/__init__.py
+-rw-r--r--  2.0 unx       58 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/parsers/__init__.py
+-rw-r--r--  2.0 unx     3715 b- defN 23-Jun-13 21:27 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/parsers/args_parser.py
+-rw-r--r--  2.0 unx     3040 b- defN 23-Jun-12 01:33 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/generators/population_generator.py
+-rw-r--r--  2.0 unx       95 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/generators/__init__.py
+-rw-r--r--  2.0 unx      389 b- defN 23-Jun-11 23:07 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/generators/char_generator.py
+-rw-r--r--  2.0 unx       50 b- defN 23-Jun-13 18:45 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/genetics/__init__.py
+-rw-r--r--  2.0 unx     3333 b- defN 23-Jun-13 17:25 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/genetics/gene.py
+-rw-r--r--  2.0 unx     1621 b- defN 23-Jun-13 18:46 hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/genetics/allele.py
+36 files, 41219 bytes uncompressed, 13178 bytes compressed:  68.0%
```

## zipnote {}

```diff
@@ -1,109 +1,109 @@
-Filename: hardyweinbergcalculator-0.2.2.post20230611/
+Filename: hardyweinbergcalculator-0.2.3.post20230613/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/PKG-INFO
+Filename: hardyweinbergcalculator-0.2.3.post20230613/PKG-INFO
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/LICENSE
+Filename: hardyweinbergcalculator-0.2.3.post20230613/LICENSE
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/pyproject.toml
+Filename: hardyweinbergcalculator-0.2.3.post20230613/pyproject.toml
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/README.md
+Filename: hardyweinbergcalculator-0.2.3.post20230613/README.md
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/setup.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/setup.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/setup.cfg
+Filename: hardyweinbergcalculator-0.2.3.post20230613/setup.cfg
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/PKG-INFO
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/PKG-INFO
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/not-zip-safe
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/not-zip-safe
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/SOURCES.txt
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/SOURCES.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/entry_points.txt
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/entry_points.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/requires.txt
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/requires.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/top_level.txt
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/top_level.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/dependency_links.txt
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/dependency_links.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/genetics/
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/genetics/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/config.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/config.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/__init__.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/__main__.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/__main__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/__init__.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/parsers/
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/parsers/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/generators/
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/generators/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/__init__.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/parsers/__init__.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/parsers/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/parsers/args_parser.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/parsers/args_parser.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/generators/population_generator.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/generators/population_generator.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/generators/__init__.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/generators/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/generators/char_generator.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/generators/char_generator.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/genetics/__init__.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/genetics/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/genetics/gene.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/genetics/gene.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/genetics/allele.py
+Filename: hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/genetics/allele.py
 Comment: 
 
 Zip file comment:
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/PKG-INFO` & `hardyweinbergcalculator-0.2.3.post20230613/PKG-INFO`

 * *Files 13% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: hardyweinbergcalculator
-Version: 0.2.2.post20230611
+Version: 0.2.3.post20230613
 Summary: Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium.
 Home-page: https://github.com/dellius-alexander/Hardy-Weinberg.git
 Author: Dellius Alexander
 Author-email: info@hyfisolutions.com
 Maintainer: info@hyfisolutions.com
 Maintainer-email: info@hyfisolutions.com
 License: MIT
@@ -38,36 +38,29 @@
 ```bash
 pip install hardyweinbergcalculator
 ```
 
 ## Usage
 
 ```text
-usage: 
-    hwc [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] [--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]]
+usage: hwc [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] [--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]] [--json JSON [JSON ...]]
 
-Alternate module usage: 
-    python3 -m hardyweinbergcalculator  [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] [--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]]
+Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium. See: https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle
 
 
-Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium. Alternate module usage: python3 -m src.main --ppop 10 --qpop 10
---pq2pop 200 --verbose See: https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle
-
 optional arguments:
-  -h, --help            show this help message and exit
-  --version             show program's version number and exit
-  --verbose             Enable verbose logging. (default: False)
-  --debug               Enable debug logging. (default: False)
-  --samples SAMPLES     Number of samples to generate, if using random data generator. (default: None)
-  --p P                 Frequency of dominant allele. (default: None)
-  --q Q                 Frequency of recessive allele. (default: None)
-  --tpop TPOP           Total population. (default: None)
-  --ppop PPOP           Original population of dominant allele. (default: None)
-  --qpop QPOP           Original population of recessive allele. (default: None)
-  --pq2pop PQ2POP       Original population of heterozygous allele. (default: None)
-  --genes GENES [GENES ...]
-                        List of JSON[Genes] to calculate. Note: This is list of json objects representing of Gene class. (default: <lambda>)
+  -h, --help         show this help message and exit
+  --version          show program's version number and exit
+  --verbose          Enable verbose logging. (default: False)
+  --debug            Enable debug logging. (default: False)
+  --samples SAMPLES  Number of samples to generate, if using random data generator. (default: None)
+  --p P              Frequency of dominant allele. (default: None)
+  --q Q              Frequency of recessive allele. (default: None)
+  --tpop TPOP        Total population. (default: None)
+  --ppop PPOP        Original population of dominant allele. (default: None)
+  --qpop QPOP        Original population of recessive allele. (default: None)
+  --pq2pop PQ2POP    Original population of heterozygous allele. (default: None)
 
-Example: python3 -m main.py --ppop 10 --qpop 10 --pq2pop 200 --verbose | Example: python3 -m [package].[module] --samples 1000 --verbose
+Example: python3 -m hwc --ppop 10 --qpop 10 --pq2pop 200 --verbose
 ```
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/LICENSE` & `hardyweinbergcalculator-0.2.3.post20230613/LICENSE`

 * *Files identical despite different names*

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/pyproject.toml` & `hardyweinbergcalculator-0.2.3.post20230613/pyproject.toml`

 * *Files 6% similar despite different names*

```diff
@@ -1,14 +1,14 @@
 [build-system]
 requires = ["setuptools>=61.0", "wheel"]
 build-backend = "setuptools.build_meta"
 
 [project]
 name = "hardyweinbergcalculator"
-version = "0.2.2"
+version = "0.2.3"
 description = 'Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium.'
 readme = {file = "README.md", content-type = "text/markdown"}
 requires-python = ">=3.7"
 license = {file = "LICENSE", content-type = "text/plain"}
 keywords = ["FastAPI", "Hardy Weinberg", "Python"]
 authors = [
   { name="Dellius Alexander",  email="info@hyfisolutions.com" },
@@ -34,15 +34,15 @@
     "sympy",
     "numpy",
     "matplotlib",
     "pandas",
 ]
 
 [project.scripts]
-hwc = "hardyweinbergcalculator.__main__:app"
+hwc = "hardyweinbergcalculator.__main__"
 
 
 [project.urls]
 "Homepage" = "https://github.com/dellius-alexander/Hardy-Weinberg.git"
 "Bug_Tracker" = "https://github.com/dellius-alexander/Hardy-Weinberg/issues"
 
 [requires]
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/README.md` & `hardyweinbergcalculator-0.2.3.post20230613/README.md`

 * *Files 13% similar despite different names*

```diff
@@ -10,34 +10,27 @@
 ```bash
 pip install hardyweinbergcalculator
 ```
 
 ## Usage
 
 ```text
-usage: 
-    hwc [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] [--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]]
+usage: hwc [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] [--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]] [--json JSON [JSON ...]]
 
-Alternate module usage: 
-    python3 -m hardyweinbergcalculator  [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] [--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]]
+Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium. See: https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle
 
 
-Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium. Alternate module usage: python3 -m src.main --ppop 10 --qpop 10
---pq2pop 200 --verbose See: https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle
-
 optional arguments:
-  -h, --help            show this help message and exit
-  --version             show program's version number and exit
-  --verbose             Enable verbose logging. (default: False)
-  --debug               Enable debug logging. (default: False)
-  --samples SAMPLES     Number of samples to generate, if using random data generator. (default: None)
-  --p P                 Frequency of dominant allele. (default: None)
-  --q Q                 Frequency of recessive allele. (default: None)
-  --tpop TPOP           Total population. (default: None)
-  --ppop PPOP           Original population of dominant allele. (default: None)
-  --qpop QPOP           Original population of recessive allele. (default: None)
-  --pq2pop PQ2POP       Original population of heterozygous allele. (default: None)
-  --genes GENES [GENES ...]
-                        List of JSON[Genes] to calculate. Note: This is list of json objects representing of Gene class. (default: <lambda>)
+  -h, --help         show this help message and exit
+  --version          show program's version number and exit
+  --verbose          Enable verbose logging. (default: False)
+  --debug            Enable debug logging. (default: False)
+  --samples SAMPLES  Number of samples to generate, if using random data generator. (default: None)
+  --p P              Frequency of dominant allele. (default: None)
+  --q Q              Frequency of recessive allele. (default: None)
+  --tpop TPOP        Total population. (default: None)
+  --ppop PPOP        Original population of dominant allele. (default: None)
+  --qpop QPOP        Original population of recessive allele. (default: None)
+  --pq2pop PQ2POP    Original population of heterozygous allele. (default: None)
 
-Example: python3 -m main.py --ppop 10 --qpop 10 --pq2pop 200 --verbose | Example: python3 -m [package].[module] --samples 1000 --verbose
+Example: python3 -m hwc --ppop 10 --qpop 10 --pq2pop 200 --verbose
 ```
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/setup.py` & `hardyweinbergcalculator-0.2.3.post20230613/setup.py`

 * *Files 4% similar despite different names*

```diff
@@ -32,15 +32,15 @@
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: Implementation :: CPython",
         "Programming Language :: Python :: Implementation :: PyPy",
     ],
     entry_points={
         "console_scripts": [
-            "hwc = hardyweinbergcalculator.__main__:app"
+            "hwc = hardyweinbergcalculator.__main__"
         ]
     },
     python_requires=">=2.7",
     include_package_data=True,
     zip_safe=False,
     install_requires=[
         "sympy",
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/setup.cfg` & `hardyweinbergcalculator-0.2.3.post20230613/setup.cfg`

 * *Files 2% similar despite different names*

```diff
@@ -1,21 +1,21 @@
 [metadata]
 name = hardyweinbergcalculator
-version = 0.2.2
+version = 0.2.3
 auther = Dellius Alexander
 auther_email = info@hyfisolutions.com
 description = 
 	A package to calculate Hardy-Weinberg Equilibrium. Hardy-Weinberg Equilibrium Calculator.
 	Calculates the expected genotype frequencies based on the allele frequencies of a population
 	in Hardy-Weinberg equilibrium.
 long_description = file: README.md
 long_description_content_type = text/markdown
 
 [egg_info]
-tag_build = .post-20230611
+tag_build = .post-20230613
 tag_date = 0
 
 [sdist]
 formats = zip
 
 [options]
 packages = 
@@ -24,9 +24,9 @@
 	sympy
 	numpy
 	matplotlib
 	pandas
 
 [options.entry_points]
 console_scripts = 
-	hwc = hardyweinbergcalculator.__main__:app
+	hwc = hardyweinbergcalculator.__main__
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/PKG-INFO` & `hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/PKG-INFO`

 * *Files 13% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: hardyweinbergcalculator
-Version: 0.2.2.post20230611
+Version: 0.2.3.post20230613
 Summary: Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium.
 Home-page: https://github.com/dellius-alexander/Hardy-Weinberg.git
 Author: Dellius Alexander
 Author-email: info@hyfisolutions.com
 Maintainer: info@hyfisolutions.com
 Maintainer-email: info@hyfisolutions.com
 License: MIT
@@ -38,36 +38,29 @@
 ```bash
 pip install hardyweinbergcalculator
 ```
 
 ## Usage
 
 ```text
-usage: 
-    hwc [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] [--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]]
+usage: hwc [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] [--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]] [--json JSON [JSON ...]]
 
-Alternate module usage: 
-    python3 -m hardyweinbergcalculator  [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] [--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]]
+Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium. See: https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle
 
 
-Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium. Alternate module usage: python3 -m src.main --ppop 10 --qpop 10
---pq2pop 200 --verbose See: https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle
-
 optional arguments:
-  -h, --help            show this help message and exit
-  --version             show program's version number and exit
-  --verbose             Enable verbose logging. (default: False)
-  --debug               Enable debug logging. (default: False)
-  --samples SAMPLES     Number of samples to generate, if using random data generator. (default: None)
-  --p P                 Frequency of dominant allele. (default: None)
-  --q Q                 Frequency of recessive allele. (default: None)
-  --tpop TPOP           Total population. (default: None)
-  --ppop PPOP           Original population of dominant allele. (default: None)
-  --qpop QPOP           Original population of recessive allele. (default: None)
-  --pq2pop PQ2POP       Original population of heterozygous allele. (default: None)
-  --genes GENES [GENES ...]
-                        List of JSON[Genes] to calculate. Note: This is list of json objects representing of Gene class. (default: <lambda>)
+  -h, --help         show this help message and exit
+  --version          show program's version number and exit
+  --verbose          Enable verbose logging. (default: False)
+  --debug            Enable debug logging. (default: False)
+  --samples SAMPLES  Number of samples to generate, if using random data generator. (default: None)
+  --p P              Frequency of dominant allele. (default: None)
+  --q Q              Frequency of recessive allele. (default: None)
+  --tpop TPOP        Total population. (default: None)
+  --ppop PPOP        Original population of dominant allele. (default: None)
+  --qpop QPOP        Original population of recessive allele. (default: None)
+  --pq2pop PQ2POP    Original population of heterozygous allele. (default: None)
 
-Example: python3 -m main.py --ppop 10 --qpop 10 --pq2pop 200 --verbose | Example: python3 -m [package].[module] --samples 1000 --verbose
+Example: python3 -m hwc --ppop 10 --qpop 10 --pq2pop 200 --verbose
 ```
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator.egg-info/SOURCES.txt` & `hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator.egg-info/SOURCES.txt`

 * *Files identical despite different names*

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/config.py` & `hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/config.py`

 * *Files identical despite different names*

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py` & `hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py`

 * *Files 4% similar despite different names*

```diff
@@ -1,9 +1,9 @@
 import json
-from ..genetics.gene import Gene
+from ..genetics import Gene
 from ..config import get_logger
 from sympy import symbols
 import numpy as np
 from typing import List
 
 log = get_logger(__name__)
 
@@ -39,14 +39,16 @@
     def __init__(self, p: float = None, q: float = None,
                  homozygous_dominant_population: float = None,
                  homozygous_recessive_population: float = None,
                  heterozygous_population: float = None,
                  total_population: float = None,
                  genes: List[Gene] = None, *args, **kwargs):
         try:
+            self.p = None
+            self.q = None
             if homozygous_dominant_population is not None \
                     and homozygous_recessive_population is not None \
                     and heterozygous_population is not None:
                 self.homozygous_dominant_population = homozygous_dominant_population
                 self.homozygous_recessive_population = homozygous_recessive_population
                 self.heterozygous_population = heterozygous_population
                 self.total_population = round(
@@ -57,68 +59,69 @@
                 self.p = (2 * self.homozygous_dominant_population + self.heterozygous_population) / \
                          (2 * self.total_population)
                 self.q = (2 * self.homozygous_recessive_population + self.heterozygous_population) / \
                          (2 * self.total_population)
                 self._calculate_expected_values()
                 self.chi_square_test = self._run_chi_square_tests()
                 self.genes = genes
-            elif p is not None or q is not None and total_population is not None:
+            elif total_population is not None and p is not None or q is not None:
                 self.total_population = round(total_population, 1)
                 self.p = p if p is not None else 1.0 - q  # p = 1 - q
                 self.q = q if q is not None else 1.0 - p  # q = 1 - p
                 self._calculate_expected_values()
                 self.chi_square_test = self._run_chi_square_tests()
                 self.genes = genes
             elif genes is not None and isinstance(genes, list) and len(genes) > 0:
                 self.genes = genes
                 # get the population count from genes
                 self.homozygous_dominant_population, \
                     self.homozygous_recessive_population, \
-                    self.heterozygous_population = self._get_population_count_from_genes(self.genes)
+                    self.heterozygous_population = HardyWeinbergStats.get_population_count_from_genes(self.genes)
                 # set the total population
                 self.total_population = round(len(genes), 1)
 
                 self.p = (2 * self.homozygous_dominant_population + self.heterozygous_population) / \
                          (2 * self.total_population)
 
                 self.q = (2 * self.homozygous_recessive_population + self.heterozygous_population) / \
                          (2 * self.total_population)
                 self._calculate_expected_values()
                 self.chi_square_test = self._run_chi_square_tests()
             else:
-                raise ValueError("Invalid parameters passed to HardyWeinbergStats constructor.\n"
-                                 "Please pass either p and total_population or homozygous_dominant_population, \n"
+                raise ValueError("\n\nInvalid parameters passed to HardyWeinbergStats constructor. Please \n"
+                                 "pass either p and total_population or homozygous_dominant_population, \n"
                                  "homozygous_recessive_population, heterozygous_population, and total_population.\n"
                                  "Please see documentation for more details.")
             self.genes = genes if genes is not None else []
         except Exception as e:
             log.error(f"Error: {e}")
 
-    def _get_population_count_from_genes(self, genes: List[Gene] = None, *args, **kwargs):
+    @staticmethod
+    def get_population_count_from_genes(genes: List[Gene], *args, **kwargs):
         """
         Get the population count from a list of genes
         :param genes:
         :param args:
         :param kwargs:
         :return:
         """
-
-        if genes is None:
-            genes = self.genes
         homozygous_dominant_population = 0
         homozygous_recessive_population = 0
         heterozygous_population = 0
-        for g in genes:
-            if g.metadata["zygous"] == "homozgous":
-                if g.metadata["homocount"] == 1 and g.metadata["dominant_trait"] is True:
-                    homozygous_dominant_population += 1
-                elif g.metadata["homocount"] == 1 and g.metadata["dominant_trait"] is False:
-                    homozygous_recessive_population += 1
-            elif g.metadata["zygous"] == "heterozygous":
-                heterozygous_population += 1
+        if genes is not None and isinstance(genes, list) and len(genes) > 0:
+            for g in genes:
+                for zygous, trait in zip(g.metadata["zygous"], g.metadata["traits"]):
+                    if zygous[0] == "homozgous":
+                        if trait[0] == "dominant":
+                            homozygous_dominant_population += 1
+                        elif trait[0] == "recessive":
+                            homozygous_recessive_population += 1
+                    elif zygous[0] == "heterozygous":
+                        heterozygous_population += 1
+
         return homozygous_dominant_population, homozygous_recessive_population, heterozygous_population
 
     def _calculate_expected_values(self):
         """
         Calculates the expected values for the Hardy-Weinberg equation
         :return: expected values
         """
@@ -151,25 +154,25 @@
             while i < len(mtx):
                 # select the observed, expected, and frequency values based
                 # on the shape of the matrix (3, 1, 3) or data structure
                 observed, expected, frequencies = mtx[0:1][0][0][i], mtx[1:2][0][0][i], mtx[2:3][0][0][i]
                 log.info(f"\nObserved: {observed} \nExpected: {expected} \nFrequency: {frequencies}")
                 chi_square_test += (((observed - expected) ** 2) / expected)
                 i += 1
-            log.info(f"Matrix shape: {mtx.shape}")
-            log.info(f"Chi-Square Test Results: {chi_square_test}")
-            log.info(f"Size of matrix: {mtx.size}")
+            log.info(f"\nMatrix shape: {mtx.shape}"
+                     f"\nChi-Square Test Results: {chi_square_test}"
+                     f"\nSize of matrix: {mtx.size}")
             return chi_square_test
         except Exception as e:
             log.error(f"Error: {e}")
 
     def __dict__(self):
         return {
-            f"{_p}": self.p,
-            f"{_q}": self.q,
+            f"p": self.p,
+            f"q": self.q,
             "expected_homozygous_dominant_population": self.p_expected,
             "expected_homozygous_recessive_population": self.q_expected,
             "expected_heterozygous_population": self.two_pq_expected,
             f"{_p} + {_q}": self.p_plus_q,
             f"{_p2}": self.p_squared,
             f"{_q2}": self.q_squared,
             f"{_2pq}": self.two_pq,
@@ -179,15 +182,15 @@
             "homozygous_recessive_population": self.homozygous_recessive_population,
             "heterozygous_population": self.heterozygous_population,
             "total_population": self.total_population,
             "@genes": [a.__dict__() for a in self.genes] if isinstance(self.genes, list) else []
         }
 
     def to_json(self):
-        return json.dumps(self.__dict__(),  indent=4, sort_keys=True)
+        return json.dumps(self.__dict__(), indent=4, sort_keys=True)
 
     def __str__(self):
         return self.to_json()
 
     def __repr__(self):
         return self.__dict__()
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py` & `hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py`

 * *Files 8% similar despite different names*

```diff
@@ -1,27 +1,29 @@
 import json
 import traceback
 from typing import List
-from ..genetics.gene import Gene
-from ..hardy_weinberg.hardy_weinberg_stats import HardyWeinbergStats
+from ..genetics import Gene
 from ..config import get_logger
+from .hardy_weinberg_stats import HardyWeinbergStats
 
 log = get_logger(__name__)
 
 
 # --------------------------------------------------------------------------- #
 class HardyWeinberg:
     stats: HardyWeinbergStats = None
 
-    def __init__(self, p: float = None, q: float = None,
+    def __init__(self,
                  homozygous_dominant_population: float = None,
                  homozygous_recessive_population: float = None,
                  heterozygous_population: float = None,
                  total_population: float = None,
-                 genes: List[Gene] = None, *args, **kwargs):
+                 genes: List[Gene] = None,
+                 p: float = None, q: float = None,
+                 *args, **kwargs):
         # log.info(f"Args: {args}")
         # log.info(f"Kwargs: {kwargs}")
         try:
             self.stats: HardyWeinbergStats = \
                 HardyWeinbergStats(p=p, q=q,
                                    total_population=total_population,
                                    homozygous_dominant_population=homozygous_dominant_population,
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/parsers/args_parser.py` & `hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/parsers/args_parser.py`

 * *Files 14% similar despite different names*

```diff
@@ -1,10 +1,12 @@
 import argparse
 import json
 import sys
+from typing import List
+
 import numpy as np
 from ...genetics.gene import Gene
 from ...genetics.allele import Allele
 from ...config import get_logger
 
 log = get_logger(__name__)
 
@@ -15,65 +17,62 @@
         prog=sys.argv[0].split('/')[-1],
         description="""
         \nHardy-Weinberg Equilibrium Calculator.
         \nCalculates the expected genotype frequencies based on the allele frequencies of a population 
         in Hardy-Weinberg equilibrium. 
       
         \nSee: https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle""",
-        usage='%(prog)s [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] ['
-              '--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]]',
+        usage='hwc [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] ['
+              '--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]] [--json JSON [JSON ...]]',
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-        epilog=("""\nExample: python3 -m %(prog)s --ppop 10 --qpop 10 --pq2pop 200 --verbose | """
-                """\nExample: python3 -m [package].[module] --samples 1000 --verbose"""),
+        epilog="""Example: python3 -m hwc --ppop 10 --qpop 10 --pq2pop 200 --verbose""",
         add_help=True,
         allow_abbrev=True,
         exit_on_error=True,
     )
-    parser.add_argument('--version', action='version', version='%(prog)s 0.1')
+    parser.add_argument('--version', action='version', version='HardyWeinberg Calculator 0.1')
     parser.add_argument('--verbose', action='store_true', default=False,
                         help='Enable verbose logging.')
     parser.add_argument('--debug', action='store_true', default=False,
                         help='Enable debug logging.')
     parser.add_argument('--samples', type=int, default=None,
-
                         help='Number of samples to generate, if using random data generator.')
     parser.add_argument('--p', type=float, default=None,
                         help='Frequency of dominant allele.')
     parser.add_argument('--q', type=float, default=None,
                         help='Frequency of recessive allele.')
     parser.add_argument('--tpop', type=float, default=None,
                         help='Total population.')
     parser.add_argument('--ppop', type=float, default=None,
                         help='Original population of dominant allele.')
     parser.add_argument('--qpop', type=float, default=None,
                         help='Original population of recessive allele.')
     parser.add_argument('--pq2pop', type=float, default=None,
                         help='Original population of heterozygous allele.')
-    parser.add_argument('--genes', type=json.loads, nargs='+', default=lambda o: o.__dict__(),
-                        help='List of JSON[Genes] to calculate. Note: This is list of json objects representing of '
-                             'Gene class.')
-
     return parser
 
 
 # --------------------------------------------------------------------------- #
-def parse_genes_from_cli(genes: list):
-    gene_list = []
+def parse_genes_from_cli(genes: List = None):
+    if genes is None:
+        log.error(f"\nGenes: {genes} is not a valid gene.")
+        return None
+    gene_list = np.array([0] * 1, dtype=Gene, ndmin=1, copy=False)
     log.info(f"\nShape of genes: {np.shape(genes)}")
     if isinstance(genes, list):
-        for gene in genes[0]:
-            # log.debug(f"\nGene: {gene}")
+        for gene in genes:
+            log.debug(f"\nGene: {gene}")
             if isinstance(gene, dict):
-                gene_list.append(Gene(mother=Allele(gene.get('mother').get('symbol')),
-                                      father=Allele(gene.get('father').get('symbol')),
-                                      description=gene.get('description')))
+                np.append(gene_list, Gene(mother=Allele(gene.get('mother').get('Alleles')[0].get('symbols')),
+                                          father=Allele(gene.get('father').get('Alleles')[0].get('symbols')),
+                                          description=gene.get('description')))
             elif isinstance(gene, Gene):
-                gene_list.append(gene)
+                np.append(gene_list, gene)
             else:
                 log.error(f"\nGene: {gene} is not a valid gene.")
     elif isinstance(genes, Gene):
-        gene_list.append(genes)
+        np.append(gene_list, genes)
     else:
-        log.error(f"\nGenes: {genes} is not a valid gene.")
-    log.debug(f"\nParsed Gene List size: {len(gene_list)}")
+        raise ValueError(f"\nGenes: {genes} is not a valid gene.",
+                         f"\nParsed Gene List size: {len(gene_list)}")
     return gene_list
 # --------------------------------------------------------------------------- #
```

## Comparing `hardyweinbergcalculator-0.2.2.post20230611/hardyweinbergcalculator/utils/generators/population_generator.py` & `hardyweinbergcalculator-0.2.3.post20230613/hardyweinbergcalculator/utils/generators/population_generator.py`

 * *Files 11% similar despite different names*

```diff
@@ -19,15 +19,15 @@
 
 
 # --------------------------------------------------------------------------- #
 def population_generator(n: int = 1):
     """Generate n number of zygotes."""
     sample = []
     for _ in range(0, n):
-        sample.append(Gene(Allele(random_chars().__next__()), Allele(random_chars().__next__())))
+        sample.append(Gene(Allele(random_chars().send(None)), Allele(random_chars().send(None))))
     yield sample
 
 
 # --------------------------------------------------------------------------- #
 async def async_population_generator(samples: int = 1):
     return population_generator(samples).__next__()
```

