# Comparing `tmp/hardyweinbergcalculator-0.2.7.post20230613.zip` & `tmp/hardyweinbergcalculator-0.2.7.post20230614.zip`

## zipinfo {}

```diff
@@ -1,38 +1,38 @@
-Zip file size: 21659 bytes, number of entries: 36
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/
--rw-r--r--  2.0 unx     2904 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/PKG-INFO
--rw-r--r--  2.0 unx     1074 b- defN 23-Jun-11 17:35 hardyweinbergcalculator-0.2.7.post20230613/LICENSE
--rw-r--r--  2.0 unx     2119 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/pyproject.toml
--rw-r--r--  2.0 unx     1567 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.7.post20230613/README.md
--rw-r--r--  2.0 unx     1976 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/setup.py
--rw-r--r--  2.0 unx      704 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/setup.cfg
--rw-r--r--  2.0 unx     2904 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/PKG-INFO
--rw-r--r--  2.0 unx        1 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/not-zip-safe
--rw-r--r--  2.0 unx     1109 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/SOURCES.txt
--rw-r--r--  2.0 unx       63 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/entry_points.txt
--rw-r--r--  2.0 unx       30 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/requires.txt
--rw-r--r--  2.0 unx       24 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/top_level.txt
--rw-r--r--  2.0 unx        1 b- defN 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/dependency_links.txt
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/
--rw-r--r--  2.0 unx      525 b- defN 23-Jun-11 17:30 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/config.py
--rw-r--r--  2.0 unx      216 b- defN 23-Jun-13 22:24 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/__init__.py
--rw-r--r--  2.0 unx     2655 b- defN 23-Jun-13 22:03 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/__main__.py
--rw-r--r--  2.0 unx     9005 b- defN 23-Jun-13 21:12 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py
--rw-r--r--  2.0 unx       95 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/__init__.py
--rw-r--r--  2.0 unx     1763 b- defN 23-Jun-13 20:07 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/parsers/
-drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-13 22:59 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/generators/
--rw-r--r--  2.0 unx      183 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/__init__.py
--rw-r--r--  2.0 unx       58 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/parsers/__init__.py
--rw-r--r--  2.0 unx     3715 b- defN 23-Jun-13 21:27 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/parsers/args_parser.py
--rw-r--r--  2.0 unx     3040 b- defN 23-Jun-12 01:33 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/generators/population_generator.py
--rw-r--r--  2.0 unx       95 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/generators/__init__.py
--rw-r--r--  2.0 unx      389 b- defN 23-Jun-11 23:07 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/generators/char_generator.py
--rw-r--r--  2.0 unx       50 b- defN 23-Jun-13 18:45 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/__init__.py
--rw-r--r--  2.0 unx     3333 b- defN 23-Jun-13 17:25 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/gene.py
--rw-r--r--  2.0 unx     1621 b- defN 23-Jun-13 18:46 hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/allele.py
-36 files, 41219 bytes uncompressed, 13147 bytes compressed:  68.1%
+Zip file size: 21027 bytes, number of entries: 36
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/
+-rw-r--r--  2.0 unx     3140 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/PKG-INFO
+-rw-r--r--  2.0 unx     1074 b- defN 23-Jun-11 17:35 hardyweinbergcalculator-0.2.7.post20230614/LICENSE
+-rw-r--r--  2.0 unx        0 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/pyproject.toml
+-rw-r--r--  2.0 unx     1803 b- defN 23-Jun-14 00:33 hardyweinbergcalculator-0.2.7.post20230614/README.md
+-rw-r--r--  2.0 unx     1976 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/setup.py
+-rw-r--r--  2.0 unx      704 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/setup.cfg
+-rw-r--r--  2.0 unx     3140 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/PKG-INFO
+-rw-r--r--  2.0 unx        1 b- defN 23-Jun-13 21:40 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/not-zip-safe
+-rw-r--r--  2.0 unx     1109 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/SOURCES.txt
+-rw-r--r--  2.0 unx       63 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/entry_points.txt
+-rw-r--r--  2.0 unx       30 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/requires.txt
+-rw-r--r--  2.0 unx       24 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/top_level.txt
+-rw-r--r--  2.0 unx        1 b- defN 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/dependency_links.txt
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/
+-rw-r--r--  2.0 unx      500 b- defN 23-Jun-13 23:26 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/config.py
+-rw-r--r--  2.0 unx      215 b- defN 23-Jun-13 23:26 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/__init__.py
+-rw-r--r--  2.0 unx     2646 b- defN 23-Jun-14 00:28 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/__main__.py
+-rw-r--r--  2.0 unx     9701 b- defN 23-Jun-13 23:32 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py
+-rw-r--r--  2.0 unx       95 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/__init__.py
+-rw-r--r--  2.0 unx     1683 b- defN 23-Jun-13 23:26 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/parsers/
+drwxr-xr-x  2.0 unx        0 b- stor 23-Jun-14 00:36 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/generators/
+-rw-r--r--  2.0 unx      183 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/__init__.py
+-rw-r--r--  2.0 unx       58 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/parsers/__init__.py
+-rw-r--r--  2.0 unx     3909 b- defN 23-Jun-13 23:26 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/parsers/args_parser.py
+-rw-r--r--  2.0 unx     3069 b- defN 23-Jun-13 23:26 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/generators/population_generator.py
+-rw-r--r--  2.0 unx       95 b- defN 23-Jun-13 18:44 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/generators/__init__.py
+-rw-r--r--  2.0 unx      390 b- defN 23-Jun-13 23:26 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/generators/char_generator.py
+-rw-r--r--  2.0 unx       50 b- defN 23-Jun-13 18:45 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/__init__.py
+-rw-r--r--  2.0 unx     3376 b- defN 23-Jun-13 23:26 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/gene.py
+-rw-r--r--  2.0 unx     1644 b- defN 23-Jun-13 23:26 hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/allele.py
+36 files, 40679 bytes uncompressed, 12515 bytes compressed:  69.2%
```

## zipnote {}

```diff
@@ -1,109 +1,109 @@
-Filename: hardyweinbergcalculator-0.2.7.post20230613/
+Filename: hardyweinbergcalculator-0.2.7.post20230614/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/PKG-INFO
+Filename: hardyweinbergcalculator-0.2.7.post20230614/PKG-INFO
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/LICENSE
+Filename: hardyweinbergcalculator-0.2.7.post20230614/LICENSE
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/pyproject.toml
+Filename: hardyweinbergcalculator-0.2.7.post20230614/pyproject.toml
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/README.md
+Filename: hardyweinbergcalculator-0.2.7.post20230614/README.md
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/setup.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/setup.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/setup.cfg
+Filename: hardyweinbergcalculator-0.2.7.post20230614/setup.cfg
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/PKG-INFO
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/PKG-INFO
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/not-zip-safe
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/not-zip-safe
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/SOURCES.txt
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/SOURCES.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/entry_points.txt
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/entry_points.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/requires.txt
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/requires.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/top_level.txt
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/top_level.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/dependency_links.txt
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/dependency_links.txt
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/config.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/config.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/__init__.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/__main__.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/__main__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/__init__.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/parsers/
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/parsers/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/generators/
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/generators/
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/__init__.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/parsers/__init__.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/parsers/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/parsers/args_parser.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/parsers/args_parser.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/generators/population_generator.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/generators/population_generator.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/generators/__init__.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/generators/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/generators/char_generator.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/generators/char_generator.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/__init__.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/__init__.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/gene.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/gene.py
 Comment: 
 
-Filename: hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/allele.py
+Filename: hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/allele.py
 Comment: 
 
 Zip file comment:
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/PKG-INFO` & `hardyweinbergcalculator-0.2.7.post20230614/PKG-INFO`

 * *Files 6% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: hardyweinbergcalculator
-Version: 0.2.7.post20230613
+Version: 0.2.7.post20230614
 Summary: Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium.
 Home-page: https://github.com/dellius-alexander/Hardy-Weinberg.git
 Author: Dellius Alexander
 Author-email: info@hyfisolutions.com
 Maintainer: info@hyfisolutions.com
 Maintainer-email: info@hyfisolutions.com
 License: MIT
@@ -18,18 +18,20 @@
 Classifier: Programming Language :: Python :: 2.7
 Classifier: Programming Language :: Python :: 3.4
 Classifier: Programming Language :: Python :: 3.5
 Classifier: Programming Language :: Python :: 3.6
 Classifier: Programming Language :: Python :: 3.7
 Classifier: Programming Language :: Python :: Implementation :: CPython
 Classifier: Programming Language :: Python :: Implementation :: PyPy
-Requires-Python: >=2.7
+Requires-Python: >=3.7
 Description-Content-Type: text/markdown
 License-File: LICENSE
 
+[![Testing FastAPI, GraphQL, MongoDB, using Docker Compose](https://github.com/dellius-alexander/Hardy-Weinberg/actions/workflows/deploy.yml/badge.svg)](https://github.com/dellius-alexander/Hardy-Weinberg/actions/workflows/deploy.yml)
+
 # Hardy Weinberg Equilibrium
 
 
 Hardy-Weinberg Equilibrium Calculator. Calculates the expected 
 genotype frequencies based on the allele frequencies of a 
 population in Hardy-Weinberg equilibrium.
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/LICENSE` & `hardyweinbergcalculator-0.2.7.post20230614/LICENSE`

 * *Files identical despite different names*

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/README.md` & `hardyweinbergcalculator-0.2.7.post20230614/README.md`

 * *Files 10% similar despite different names*

```diff
@@ -1,7 +1,9 @@
+[![Testing FastAPI, GraphQL, MongoDB, using Docker Compose](https://github.com/dellius-alexander/Hardy-Weinberg/actions/workflows/deploy.yml/badge.svg)](https://github.com/dellius-alexander/Hardy-Weinberg/actions/workflows/deploy.yml)
+
 # Hardy Weinberg Equilibrium
 
 
 Hardy-Weinberg Equilibrium Calculator. Calculates the expected 
 genotype frequencies based on the allele frequencies of a 
 population in Hardy-Weinberg equilibrium.
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/setup.py` & `hardyweinbergcalculator-0.2.7.post20230614/setup.py`

 * *Files 0% similar despite different names*

```diff
@@ -34,15 +34,15 @@
         "Programming Language :: Python :: Implementation :: PyPy",
     ],
     entry_points={
         "console_scripts": [
             "hwc = hardyweinbergcalculator.__main__:main"
         ]
     },
-    python_requires=">=2.7",
+    python_requires=">=3.7",
     include_package_data=True,
     zip_safe=False,
     install_requires=[
         "sympy",
         "numpy",
         "matplotlib",
         "pandas",
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/setup.cfg` & `hardyweinbergcalculator-0.2.7.post20230614/setup.cfg`

 * *Files 1% similar despite different names*

```diff
@@ -7,15 +7,15 @@
 	A package to calculate Hardy-Weinberg Equilibrium. Hardy-Weinberg Equilibrium Calculator.
 	Calculates the expected genotype frequencies based on the allele frequencies of a population
 	in Hardy-Weinberg equilibrium.
 long_description = file: README.md
 long_description_content_type = text/markdown
 
 [egg_info]
-tag_build = .post-20230613
+tag_build = .post-20230614
 tag_date = 0
 
 [sdist]
 formats = zip
 
 [options]
 packages =
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/PKG-INFO` & `hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/PKG-INFO`

 * *Files 6% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: hardyweinbergcalculator
-Version: 0.2.7.post20230613
+Version: 0.2.7.post20230614
 Summary: Hardy-Weinberg Equilibrium Calculator. Calculates the expected genotype frequencies based on the allele frequencies of a population in Hardy-Weinberg equilibrium.
 Home-page: https://github.com/dellius-alexander/Hardy-Weinberg.git
 Author: Dellius Alexander
 Author-email: info@hyfisolutions.com
 Maintainer: info@hyfisolutions.com
 Maintainer-email: info@hyfisolutions.com
 License: MIT
@@ -18,18 +18,20 @@
 Classifier: Programming Language :: Python :: 2.7
 Classifier: Programming Language :: Python :: 3.4
 Classifier: Programming Language :: Python :: 3.5
 Classifier: Programming Language :: Python :: 3.6
 Classifier: Programming Language :: Python :: 3.7
 Classifier: Programming Language :: Python :: Implementation :: CPython
 Classifier: Programming Language :: Python :: Implementation :: PyPy
-Requires-Python: >=2.7
+Requires-Python: >=3.7
 Description-Content-Type: text/markdown
 License-File: LICENSE
 
+[![Testing FastAPI, GraphQL, MongoDB, using Docker Compose](https://github.com/dellius-alexander/Hardy-Weinberg/actions/workflows/deploy.yml/badge.svg)](https://github.com/dellius-alexander/Hardy-Weinberg/actions/workflows/deploy.yml)
+
 # Hardy Weinberg Equilibrium
 
 
 Hardy-Weinberg Equilibrium Calculator. Calculates the expected 
 genotype frequencies based on the allele frequencies of a 
 population in Hardy-Weinberg equilibrium.
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator.egg-info/SOURCES.txt` & `hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator.egg-info/SOURCES.txt`

 * *Files identical despite different names*

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/__main__.py` & `hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/__main__.py`

 * *Files 3% similar despite different names*

```diff
@@ -1,36 +1,40 @@
 import logging
 import sys
 import traceback
 from _ctypes import Array
 from typing import Union, List
-from . import Gene, HardyWeinberg, generate_population, parse_args, get_logger, parse_genes_from_cli
+from . import Gene, HardyWeinberg, generate_population, parse_args, get_logger
 
 log = get_logger(__name__)
 
 
 # --------------------------------------------------------------------------- #
 def app(
-        total_population: float = None,
-        homozygous_dominant_population: float = None,
-        homozygous_recessive_population: float = None,
-        heterozygous_population: float = None,
-        genes: Union[List[Gene], Array[Gene]] = None,
-        **kwargs
+    total_population: float = None,
+    homozygous_dominant_population: float = None,
+    homozygous_recessive_population: float = None,
+    heterozygous_population: float = None,
+    genes: Union[List[Gene]] = None,
+    **kwargs,
 ):
     try:  # check for None inputs
-        if total_population is None and genes is None and heterozygous_population is None:
+        if (
+            total_population is None
+            and genes is None
+            and heterozygous_population is None
+        ):
             raise ValueError("Genes or population data must be provided.")
         return HardyWeinberg(
             total_population=total_population,
             homozygous_dominant_population=homozygous_dominant_population,
             homozygous_recessive_population=homozygous_recessive_population,
             heterozygous_population=heterozygous_population,
             genes=genes,
-            **kwargs
+            **kwargs,
         )
     except Exception as e:
         log.error(e)
         log.error(traceback.format_exc())
 
 
 # --------------------------------------------------------------------------- #
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py` & `hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg_stats.py`

 * *Files 5% similar despite different names*

```diff
@@ -5,15 +5,15 @@
 import numpy as np
 from typing import List
 
 log = get_logger(__name__)
 
 # --------------------------------------------------------------------------- #
 # Define symbols for Hardy-Weinberg equations
-_p, _q, _p2, _2pq, _q2 = symbols('p, q, p**2, 2*pq, q**2')
+_p, _q, _p2, _2pq, _q2 = symbols("p, q, p**2, 2*pq, q**2")
 
 
 # --------------------------------------------------------------------------- #
 class HardyWeinbergStats:
     """Hardy-Weinberg Statistics class:
     defines a single Hardy-Weinberg Statistics object.
     """
@@ -32,69 +32,94 @@
     homozygous_recessive_population: float = None
     heterozygous_population: float = None
     p_squared_plus_two_pq_plus_q_squared: float = None
     chi_square_test: float = None
     genes: List[Gene] = None
     data_structure: np.array = None
 
-    def __init__(self, p: float = None, q: float = None,
-                 homozygous_dominant_population: float = None,
-                 homozygous_recessive_population: float = None,
-                 heterozygous_population: float = None,
-                 total_population: float = None,
-                 genes: List[Gene] = None, *args, **kwargs):
+    def __init__(
+        self,
+        p: float = None,
+        q: float = None,
+        homozygous_dominant_population: float = None,
+        homozygous_recessive_population: float = None,
+        heterozygous_population: float = None,
+        total_population: float = None,
+        genes: List[Gene] = None,
+        *args,
+        **kwargs,
+    ):
         try:
             self.p = None
             self.q = None
-            if homozygous_dominant_population is not None \
-                    and homozygous_recessive_population is not None \
-                    and heterozygous_population is not None:
+            if (
+                homozygous_dominant_population is not None
+                and homozygous_recessive_population is not None
+                and heterozygous_population is not None
+            ):
                 self.homozygous_dominant_population = homozygous_dominant_population
                 self.homozygous_recessive_population = homozygous_recessive_population
                 self.heterozygous_population = heterozygous_population
                 self.total_population = round(
-                    sum([self.homozygous_dominant_population,
-                         self.homozygous_recessive_population,
-                         self.heterozygous_population]),
-                    1)
-                self.p = (2 * self.homozygous_dominant_population + self.heterozygous_population) / \
-                         (2 * self.total_population)
-                self.q = (2 * self.homozygous_recessive_population + self.heterozygous_population) / \
-                         (2 * self.total_population)
+                    sum(
+                        [
+                            self.homozygous_dominant_population,
+                            self.homozygous_recessive_population,
+                            self.heterozygous_population,
+                        ]
+                    ),
+                    1,
+                )
+                self.p = (
+                    2 * self.homozygous_dominant_population
+                    + self.heterozygous_population
+                ) / (2 * self.total_population)
+                self.q = (
+                    2 * self.homozygous_recessive_population
+                    + self.heterozygous_population
+                ) / (2 * self.total_population)
                 self._calculate_expected_values()
                 self.chi_square_test = self._run_chi_square_tests()
                 self.genes = genes
             elif total_population is not None and p is not None or q is not None:
                 self.total_population = round(total_population, 1)
                 self.p = p if p is not None else 1.0 - q  # p = 1 - q
                 self.q = q if q is not None else 1.0 - p  # q = 1 - p
                 self._calculate_expected_values()
                 self.chi_square_test = self._run_chi_square_tests()
                 self.genes = genes
             elif genes is not None and isinstance(genes, list) and len(genes) > 0:
                 self.genes = genes
                 # get the population count from genes
-                self.homozygous_dominant_population, \
-                    self.homozygous_recessive_population, \
-                    self.heterozygous_population = HardyWeinbergStats.get_population_count_from_genes(self.genes)
+                (
+                    self.homozygous_dominant_population,
+                    self.homozygous_recessive_population,
+                    self.heterozygous_population,
+                ) = HardyWeinbergStats.get_population_count_from_genes(self.genes)
                 # set the total population
                 self.total_population = round(len(genes), 1)
 
-                self.p = (2 * self.homozygous_dominant_population + self.heterozygous_population) / \
-                         (2 * self.total_population)
-
-                self.q = (2 * self.homozygous_recessive_population + self.heterozygous_population) / \
-                         (2 * self.total_population)
+                self.p = (
+                    2 * self.homozygous_dominant_population
+                    + self.heterozygous_population
+                ) / (2 * self.total_population)
+
+                self.q = (
+                    2 * self.homozygous_recessive_population
+                    + self.heterozygous_population
+                ) / (2 * self.total_population)
                 self._calculate_expected_values()
                 self.chi_square_test = self._run_chi_square_tests()
             else:
-                raise ValueError("\n\nInvalid parameters passed to HardyWeinbergStats constructor. Please \n"
-                                 "pass either p and total_population or homozygous_dominant_population, \n"
-                                 "homozygous_recessive_population, heterozygous_population, and total_population.\n"
-                                 "Please see documentation for more details.")
+                raise ValueError(
+                    "\n\nInvalid parameters passed to HardyWeinbergStats constructor. Please \n"
+                    "pass either p and total_population or homozygous_dominant_population, \n"
+                    "homozygous_recessive_population, heterozygous_population, and total_population.\n"
+                    "Please see documentation for more details."
+                )
             self.genes = genes if genes is not None else []
         except Exception as e:
             log.error(f"Error: {e}")
 
     @staticmethod
     def get_population_count_from_genes(genes: List[Gene], *args, **kwargs):
         """
@@ -114,86 +139,107 @@
                         if trait[0] == "dominant":
                             homozygous_dominant_population += 1
                         elif trait[0] == "recessive":
                             homozygous_recessive_population += 1
                     elif zygous[0] == "heterozygous":
                         heterozygous_population += 1
 
-        return homozygous_dominant_population, homozygous_recessive_population, heterozygous_population
+        return (
+            homozygous_dominant_population,
+            homozygous_recessive_population,
+            heterozygous_population,
+        )
 
     def _calculate_expected_values(self):
         """
         Calculates the expected values for the Hardy-Weinberg equation
         :return: expected values
         """
-        self.p_expected = (self.p ** 2) * self.total_population
-        self.q_expected = (self.q ** 2) * self.total_population
+        self.p_expected = (self.p**2) * self.total_population
+        self.q_expected = (self.q**2) * self.total_population
         self.two_pq_expected = (2 * self.p * self.q) * self.total_population
         self.p_plus_q = self.p + self.q
-        self.p_squared = self.p ** 2
-        self.q_squared = self.q ** 2
+        self.p_squared = self.p**2
+        self.q_squared = self.q**2
         self.two_pq = 2 * self.p * self.q
-        self.p_squared_plus_two_pq_plus_q_squared = self.p_squared + self.two_pq + self.q_squared
+        self.p_squared_plus_two_pq_plus_q_squared = (
+            self.p_squared + self.two_pq + self.q_squared
+        )
 
     def _run_chi_square_tests(self, *args, **kwargs):
         """
         Performs the chi-square test on the observed and expected values
         :return: chi-square test results
         """
         try:
             mtx = np.array(
                 object=[
-                    [[self.homozygous_dominant_population, self.homozygous_recessive_population, self.heterozygous_population]],
+                    [
+                        [
+                            self.homozygous_dominant_population,
+                            self.homozygous_recessive_population,
+                            self.heterozygous_population,
+                        ]
+                    ],
                     [[self.p_expected, self.q_expected, self.two_pq_expected]],
                     [[self.p_squared, self.q_squared, self.two_pq]],
                 ],
                 dtype=np.float64,
             )
             log.info(f"\nStats Matrices: \n{mtx}")
             i = 0
             chi_square_test = 0.0
             while i < len(mtx):
                 # select the observed, expected, and frequency values based
                 # on the shape of the matrix (3, 1, 3) or data structure
-                observed, expected, frequencies = mtx[0:1][0][0][i], mtx[1:2][0][0][i], mtx[2:3][0][0][i]
-                log.info(f"\nObserved: {observed} \nExpected: {expected} \nFrequency: {frequencies}")
-                chi_square_test += (((observed - expected) ** 2) / expected)
+                observed, expected, frequencies = (
+                    mtx[0:1][0][0][i],
+                    mtx[1:2][0][0][i],
+                    mtx[2:3][0][0][i],
+                )
+                log.info(
+                    f"\nObserved: {observed} \nExpected: {expected} \nFrequency: {frequencies}"
+                )
+                chi_square_test += ((observed - expected) ** 2) / expected
                 i += 1
-            log.info(f"\nMatrix shape: {mtx.shape}"
-                     f"\nChi-Square Test Results: {chi_square_test}"
-                     f"\nSize of matrix: {mtx.size}")
+            log.info(
+                f"\nMatrix shape: {mtx.shape}"
+                f"\nChi-Square Test Results: {chi_square_test}"
+                f"\nSize of matrix: {mtx.size}"
+            )
             return chi_square_test
         except Exception as e:
             log.error(f"Error: {e}")
 
     def __dict__(self):
         return {
-            f"p": self.p,
-            f"q": self.q,
+            f"{_p}": self.p,
+            f"{_q}": self.q,
             "expected_homozygous_dominant_population": self.p_expected,
             "expected_homozygous_recessive_population": self.q_expected,
             "expected_heterozygous_population": self.two_pq_expected,
             f"{_p} + {_q}": self.p_plus_q,
             f"{_p2}": self.p_squared,
             f"{_q2}": self.q_squared,
             f"{_2pq}": self.two_pq,
             f"{_p2} + {_2pq} + {_q2}": self.p_squared_plus_two_pq_plus_q_squared,
             "chi_square_test": self.chi_square_test,
             "homozygous_dominant_population": self.homozygous_dominant_population,
             "homozygous_recessive_population": self.homozygous_recessive_population,
             "heterozygous_population": self.heterozygous_population,
             "total_population": self.total_population,
-            "@genes": [a.__dict__() for a in self.genes] if isinstance(self.genes, list) else []
+            "@genes": [a.__dict__() for a in self.genes]
+            if isinstance(self.genes, list)
+            else [],
         }
 
     def to_json(self):
         return json.dumps(self.__dict__(), indent=4, sort_keys=True)
 
     def __str__(self):
         return self.to_json()
 
     def __repr__(self):
         return self.__dict__()
 
     def __iter__(self):
         return iter(self.__dict__())
-
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py` & `hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/hardy_weinberg/hardy_weinberg.py`

 * *Files 10% similar despite different names*

```diff
@@ -8,42 +8,50 @@
 log = get_logger(__name__)
 
 
 # --------------------------------------------------------------------------- #
 class HardyWeinberg:
     stats: HardyWeinbergStats = None
 
-    def __init__(self,
-                 homozygous_dominant_population: float = None,
-                 homozygous_recessive_population: float = None,
-                 heterozygous_population: float = None,
-                 total_population: float = None,
-                 genes: List[Gene] = None,
-                 p: float = None, q: float = None,
-                 *args, **kwargs):
+    def __init__(
+        self,
+        homozygous_dominant_population: float = None,
+        homozygous_recessive_population: float = None,
+        heterozygous_population: float = None,
+        total_population: float = None,
+        genes: List[Gene] = None,
+        p: float = None,
+        q: float = None,
+        *args,
+        **kwargs,
+    ):
         # log.info(f"Args: {args}")
         # log.info(f"Kwargs: {kwargs}")
         try:
-            self.stats: HardyWeinbergStats = \
-                HardyWeinbergStats(p=p, q=q,
-                                   total_population=total_population,
-                                   homozygous_dominant_population=homozygous_dominant_population,
-                                   homozygous_recessive_population=homozygous_recessive_population,
-                                   heterozygous_population=heterozygous_population,
-                                   genes=genes,
-                                   *args, **kwargs)
+            self.stats: HardyWeinbergStats = HardyWeinbergStats(
+                p=p,
+                q=q,
+                total_population=total_population,
+                homozygous_dominant_population=homozygous_dominant_population,
+                homozygous_recessive_population=homozygous_recessive_population,
+                heterozygous_population=heterozygous_population,
+                genes=genes,
+                *args,
+                **kwargs,
+            )
         except Exception as e:
             log.error(e)
             log.error(traceback.format_exc())
 
     def __dict__(self):
         return self.stats.__dict__()
 
     def __repr__(self):
         return self.stats.__repr__()
 
     def __str__(self):
         return self.stats.__str__()
 
     def to_json(self):
-        return json.dumps(self.__dict__(), default=lambda o: o.__dict__, indent=4, sort_keys=False)
-
+        return json.dumps(
+            self.__dict__(), default=lambda o: o.__dict__, indent=4, sort_keys=False
+        )
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/parsers/args_parser.py` & `hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/parsers/args_parser.py`

 * *Files 26% similar despite different names*

```diff
@@ -10,69 +10,103 @@
 
 log = get_logger(__name__)
 
 
 # --------------------------------------------------------------------------- #
 def parse_args():
     parser = argparse.ArgumentParser(
-        prog=sys.argv[0].split('/')[-1],
+        prog=sys.argv[0].split("/")[-1],
         description="""
         \nHardy-Weinberg Equilibrium Calculator.
         \nCalculates the expected genotype frequencies based on the allele frequencies of a population 
         in Hardy-Weinberg equilibrium. 
       
         \nSee: https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle""",
-        usage='hwc [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] ['
-              '--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]] [--json JSON [JSON ...]]',
+        usage="hwc [-h] [--version] [--verbose] [--debug] [--samples SAMPLES] [--p P] [--q Q] [--tpop TPOP] ["
+        "--ppop PPOP] [--qpop QPOP] [--pq2pop PQ2POP] [--genes GENES [GENES ...]] [--json JSON [JSON ...]]",
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
         epilog="""Example: python3 -m hwc --ppop 10 --qpop 10 --pq2pop 200 --verbose""",
         add_help=True,
         allow_abbrev=True,
         exit_on_error=True,
     )
-    parser.add_argument('--version', action='version', version='HardyWeinberg Calculator 0.1')
-    parser.add_argument('--verbose', action='store_true', default=False,
-                        help='Enable verbose logging.')
-    parser.add_argument('--debug', action='store_true', default=False,
-                        help='Enable debug logging.')
-    parser.add_argument('--samples', type=int, default=None,
-                        help='Number of samples to generate, if using random data generator.')
-    parser.add_argument('--p', type=float, default=None,
-                        help='Frequency of dominant allele.')
-    parser.add_argument('--q', type=float, default=None,
-                        help='Frequency of recessive allele.')
-    parser.add_argument('--tpop', type=float, default=None,
-                        help='Total population.')
-    parser.add_argument('--ppop', type=float, default=None,
-                        help='Original population of dominant allele.')
-    parser.add_argument('--qpop', type=float, default=None,
-                        help='Original population of recessive allele.')
-    parser.add_argument('--pq2pop', type=float, default=None,
-                        help='Original population of heterozygous allele.')
+    parser.add_argument(
+        "--version", action="version", version="HardyWeinberg Calculator 0.1"
+    )
+    parser.add_argument(
+        "--verbose", action="store_true", default=False, help="Enable verbose logging."
+    )
+    parser.add_argument(
+        "--debug", action="store_true", default=False, help="Enable debug logging."
+    )
+    parser.add_argument(
+        "--samples",
+        type=int,
+        default=None,
+        help="Number of samples to generate, if using random data generator.",
+    )
+    parser.add_argument(
+        "--p", type=float, default=None, help="Frequency of dominant allele."
+    )
+    parser.add_argument(
+        "--q", type=float, default=None, help="Frequency of recessive allele."
+    )
+    parser.add_argument("--tpop", type=float, default=None, help="Total population.")
+    parser.add_argument(
+        "--ppop",
+        type=float,
+        default=None,
+        help="Original population of dominant allele.",
+    )
+    parser.add_argument(
+        "--qpop",
+        type=float,
+        default=None,
+        help="Original population of recessive allele.",
+    )
+    parser.add_argument(
+        "--pq2pop",
+        type=float,
+        default=None,
+        help="Original population of heterozygous allele.",
+    )
     return parser
 
 
 # --------------------------------------------------------------------------- #
 def parse_genes_from_cli(genes: List = None):
     if genes is None:
         log.error(f"\nGenes: {genes} is not a valid gene.")
         return None
     gene_list = np.array([0] * 1, dtype=Gene, ndmin=1, copy=False)
     log.info(f"\nShape of genes: {np.shape(genes)}")
     if isinstance(genes, list):
         for gene in genes:
             log.debug(f"\nGene: {gene}")
             if isinstance(gene, dict):
-                np.append(gene_list, Gene(mother=Allele(gene.get('mother').get('Alleles')[0].get('symbols')),
-                                          father=Allele(gene.get('father').get('Alleles')[0].get('symbols')),
-                                          description=gene.get('description')))
+                np.append(
+                    gene_list,
+                    Gene(
+                        mother=Allele(
+                            gene.get("mother").get("Alleles")[0].get("symbols")
+                        ),
+                        father=Allele(
+                            gene.get("father").get("Alleles")[0].get("symbols")
+                        ),
+                        description=gene.get("description"),
+                    ),
+                )
             elif isinstance(gene, Gene):
                 np.append(gene_list, gene)
             else:
                 log.error(f"\nGene: {gene} is not a valid gene.")
     elif isinstance(genes, Gene):
         np.append(gene_list, genes)
     else:
-        raise ValueError(f"\nGenes: {genes} is not a valid gene.",
-                         f"\nParsed Gene List size: {len(gene_list)}")
+        raise ValueError(
+            f"\nGenes: {genes} is not a valid gene.",
+            f"\nParsed Gene List size: {len(gene_list)}",
+        )
     return gene_list
+
+
 # --------------------------------------------------------------------------- #
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/utils/generators/population_generator.py` & `hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/utils/generators/population_generator.py`

 * *Files 8% similar despite different names*

```diff
@@ -19,15 +19,17 @@
 
 
 # --------------------------------------------------------------------------- #
 def population_generator(n: int = 1):
     """Generate n number of zygotes."""
     sample = []
     for _ in range(0, n):
-        sample.append(Gene(Allele(random_chars().send(None)), Allele(random_chars().send(None))))
+        sample.append(
+            Gene(Allele(random_chars().send(None)), Allele(random_chars().send(None)))
+        )
     yield sample
 
 
 # --------------------------------------------------------------------------- #
 async def async_population_generator(samples: int = 1):
     return population_generator(samples).__next__()
 
@@ -41,28 +43,32 @@
     # then we divide the number of samples by the lcm to get the chunk size
     chunk_size = samples // lcm if samples > lcm else samples
     # then we divide the number of samples by the chunk size to get the number of chunks
     samples_per_chunk = samples // chunk_size
     # then we create a container to hold the samples
     samples_container = []
     # print the info
-    log.info(f"\nSamples: {samples}"
-             f"\nLeast Common Multiple: {lcm}"
-             f"\nChunk Size: {chunk_size}"
-             f"\nSamples per Chunk: {samples_per_chunk}")
+    log.info(
+        f"\nSamples: {samples}"
+        f"\nLeast Common Multiple: {lcm}"
+        f"\nChunk Size: {chunk_size}"
+        f"\nSamples per Chunk: {samples_per_chunk}"
+    )
 
     # lastly we iterate over the chunk size and append the chunked samples to the container
     while len(samples_container) < samples:
         task = asyncio.create_task(async_population_generator(samples_per_chunk))
         tasks.append(task)
         res = await asyncio.gather(*tasks)
         samples_container.extend(res[0])
     end = perf_counter()
-    log.info(f"\nTime Elapsed: {end - start:0.4f} seconds"
-              f"\nNext Chunk Size: {len(samples_container)}")
+    log.info(
+        f"\nTime Elapsed: {end - start:0.4f} seconds"
+        f"\nNext Chunk Size: {len(samples_container)}"
+    )
     return samples_container
 
 
 # --------------------------------------------------------------------------- #
 async def async_population_generator_handler(samples: int, **kwargs) -> List[Gene]:
     try:
         res = await async_chunked_population_generator(samples)
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/gene.py` & `hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/gene.py`

 * *Files 1% similar despite different names*

```diff
@@ -14,25 +14,28 @@
     that contain one allele each.  A diploid contains two alleles, one from
     each parent.
 
     - mother: Allele - first allele
     - father: Allele - second allele
     - description: str - description of the gene
     """
+
     mother: Allele = None  # first allele
     father: Allele = None  # second allele
     description: str = None
     matrix: np.array = None
     metadata: dict = dict()
 
     def __init__(self, mother: Allele, father: Allele, description: str = None):
         self.mother = mother
         self.father = father
         self.description = description
-        self.matrix = np.array(list(product(self.mother.symbols, self.father.symbols)), dtype=object)
+        self.matrix = np.array(
+            list(product(self.mother.symbols, self.father.symbols)), dtype=object
+        )
         self.metadata = dict()
         self.metadata = self._get_metadata()
 
     def _get_metadata(self):
         # get metadata about the gene's alleles
         metadata = dict()
         metadata["dominant_traits"] = 0
@@ -82,13 +85,13 @@
     def __next__(self):
         return self.__dict__()
 
     def __repr__(self):
         return self.__str__()
 
     def to_json(self):
-        return json.dumps(self.__dict__(), default=lambda o: o.__dict__, indent=4, sort_keys=True)
+        return json.dumps(
+            self.__dict__(), default=lambda o: o.__dict__, indent=4, sort_keys=True
+        )
 
     def __str__(self):
         return str(self.__dict__())
-
-
```

## Comparing `hardyweinbergcalculator-0.2.7.post20230613/hardyweinbergcalculator/genetics/allele.py` & `hardyweinbergcalculator-0.2.7.post20230614/hardyweinbergcalculator/genetics/allele.py`

 * *Files 3% similar despite different names*

```diff
@@ -8,20 +8,20 @@
 # --------------------------------------------------------------------------- #
 class Allele:
     """Allele class:
     defines a single allele object.
 
     :param symbols: List - the unicode or ascii symbols for the allele. e.g. 'A' or 'a'
     """
+
     _symbols: List
     _traits: List
     _matrix: np.array
 
     def __init__(self, symbols: List = random_chars().__next__()):
-
         self._symbols = symbols
         self._traits = []
         for symbol in self._symbols:
             if symbol.isupper():
                 self._traits.append([f"{symbol}", "dominant"])
             elif symbol.islower():
                 self._traits.append([f"{symbol}", "recessive"])
@@ -52,13 +52,16 @@
     def __next__(self):
         return self.__dict__()
 
     def __repr__(self):
         return self.__dict__()
 
     def to_json(self):
-        return json.dumps(self.__dict__(), default=lambda o: o.__dict__(), indent=4, sort_keys=True)
+        return json.dumps(
+            self.__dict__(), default=lambda o: o.__dict__(), indent=4, sort_keys=True
+        )
 
     def __str__(self):
         return self.__dict__()
 
+
 # --------------------------------------------------------------------------- #
```

